### PR TITLE
fix: add query params

### PR DESCRIPTION
**Pull request by tenadopt**
Add query params to the URL.

### CODE REVIEW

File: src/features/imageEditor/ui/CanvasEditor.tsx [hunk 1]
```diff
@@ -1,12 +1,10 @@
-import React, { useRef, useState, useEffect, forwardRef, useImperativeHandle } from 'react';
-import { useDispatch, useSelector } from 'react-redux';
-import { RootState } from 'app/store';
+import React, {useRef, useState, useEffect, forwardRef, useImperativeHandle} from 'react';
+import {useSearchParams} from 'react-router-dom';
```
Comment (shev7) [inline src/features/imageEditor/ui/CanvasEditor.tsx:2]: not formatted with prettier, seems like precommit is not working

File: src/features/imageEditor/ui/CanvasEditor.tsx [hunk 2]
```diff
@@ -22,17 +20,19 @@ export interface CanvasEditorHandle {
...
-const CanvasEditor = forwardRef<CanvasEditorHandle, CanvasEditorProps>(({ imageUrl, onSave, loadImage }, ref) => {
-    const dispatch = useDispatch();
-    const settings = useSelector((state: RootState) => state.imageEditor.settings);
+const CanvasEditor = forwardRef<CanvasEditorHandle, CanvasEditorProps>(({imageUrl, onSave, loadImage}, ref) => {
+    const [searchParams, setSearchParams] = useSearchParams();
     const canvasRef = useRef<HTMLCanvasElement | null>(null);
     const contextRef = useRef<CanvasRenderingContext2D | null>(null);
     const [isDrawing, setIsDrawing] = useState(false);
-    const [shape, setShape] = useState<ShapeType>('line');
+    const [shape, setShape] = useState<ShapeType>(searchParams.get('tool') as ShapeType || 'line');
```
Comment (shev7) [inline src/features/imageEditor/ui/CanvasEditor.tsx:28]: you don't even need local state, just use value returned from searchParams.get()

File: src/features/imageEditor/ui/CanvasEditor.tsx [hunk 3]
```diff
@@ -65,6 +65,9 @@ const CanvasEditor = forwardRef<CanvasEditorHandle, CanvasEditorProps>(({ imageU
         canvas.style.height = `${window.innerHeight * 0.8}px`;
 
         updateCanvasContext();
+
+        setSearchParams({...Object.fromEntries(searchParams.entries()), tool: shape, color, sizeBrush: brushSize.toString() });
```
Comment (shev7) [inline src/features/imageEditor/ui/CanvasEditor.tsx:69]:
```suggestion
        setSearchParams({...Object.fromEntries(searchParams.entries()), tool: shape, color, brushSize: brushSize.toString() });
```

File: src/features/imageEditor/ui/CanvasEditor.tsx [hunk 4]
```diff
@@ -350,11 +347,11 @@ const CanvasEditor = forwardRef<CanvasEditorHandle, CanvasEditorProps>(({ imageU
                     onMouseUp={finishDrawing}
                     onMouseMove={draw}
                     onMouseLeave={finishDrawing}
-                    style={{ border: '1px solid #000' }}
+                    style={{border: '1px solid #000'}}
```
Comment (shev7) [inline src/features/imageEditor/ui/CanvasEditor.tsx:350]: avoid using inline styles

File: src/pages/ImageEditor/ui/ImageEditorPage.tsx [hunk 5]
```diff
@@ -166,7 +169,7 @@ const ImageEditorPage = () => {
                 </Button>
             </Box>
             <Box mt={4}>
-                <CanvasEditor ref={canvasEditorRef} imageUrl={imageUrl} onSave={setImageUrl} loadImage={loadImage} />
+                <CanvasEditor ref={canvasEditorRef} imageUrl={imageUrl} onSave={setImageUrl} loadImage={loadImage} {...queryParams} />
```
Comment (shev7) [inline src/pages/ImageEditor/ui/ImageEditorPage.tsx:172]: why do you pass those params, if you are not using them from props??
you don't even need to pass them, just handle query params internally in CanvasEditor

File: src/features/imageEditor/ui/CanvasEditor.tsx [hunk 6]
```diff
@@ -328,19 +325,19 @@ const CanvasEditor = forwardRef<CanvasEditorHandle, CanvasEditorProps>(({ imageU
...
                 />
                 <Slider
-                    value={settings.brushSize}
+                    value={brushSize}
                     onChange={(_, newValue) => handleBrushSizeChange(_, newValue)}
                     min={1}
                     max={50}
                     valueLabelDisplay="auto"
-                    style={{ width: 200, margin: '0 20px' }}
+                    style={{width: 200, margin: '0 20px'}}
```
Comment (shev7) [inline src/features/imageEditor/ui/CanvasEditor.tsx:340]: aboid

**Review by shev7 — CHANGES_REQUESTED** [review]

File: src/features/imageEditor/ui/CanvasEditor.tsx (hunk 2 above)
Comment (tenadopt) [reply src/features/imageEditor/ui/CanvasEditor.tsx:28]: Ok

File: src/pages/ImageEditor/ui/ImageEditorPage.tsx (hunk 5 above)
Comment (tenadopt) [reply src/pages/ImageEditor/ui/ImageEditorPage.tsx:172]: You are right

File: src/features/imageEditor/ui/CanvasEditor.tsx (hunk 4 above)
Comment (tenadopt) [reply src/features/imageEditor/ui/CanvasEditor.tsx:350]: Remove

File: src/features/imageEditor/ui/CanvasEditor.tsx (hunk 6 above)
Comment (tenadopt) [reply src/features/imageEditor/ui/CanvasEditor.tsx:340]: Remove

**Pull request closed without merging**